### PR TITLE
cni ambient config dir hostpath support setting by user

### PR DESCRIPTION
**Pull request by rokkiter**
**Please provide a description of this PR:**

Fixes(#45400)

### CODE REVIEW

File: manifests/charts/istio-cni/values.yaml [hunk 1]
```diff
@@ -19,6 +19,7 @@ cni:
   # defaults:
   cniBinDir: "" # Auto-detected based on version; defaults to /opt/cni/bin.
   cniConfDir: /etc/cni/net.d
+  cniAmbientConfigDir: "" # defaults to /etc/ambient-config
```
Comment (hanxiaop) [inline manifests/charts/istio-cni/values.yaml:22]: Should we keep this in the ambient setting field?

File: operator/pkg/apis/istio/v1alpha1/values_types.proto [hunk 2]
```diff
@@ -60,21 +60,21 @@ message CNIConfig {
 
   string cniConfDir = 7;
 
-  string cniConfFileName = 8;
+  string cniAmbientConfigDir = 8;
```
Comment (hanxiaop) [inline operator/pkg/apis/istio/v1alpha1/values_types.proto:63]: Why not append the new field to the last? This may have compatibility issues.

Comment (rokkiter) [reply operator/pkg/apis/istio/v1alpha1/values_types.proto]: thinks，It has been modified

File: manifests/charts/istio-cni/values.yaml (hunk 1 above)
Comment (rokkiter) [reply manifests/charts/istio-cni/values.yaml]: thanks, I had moved it in ambient setting

File: manifests/charts/istio-cni/values.yaml [hunk 3]
```diff
@@ -54,6 +54,9 @@ cni:
     enabled: false
     # Set ambient redirection mode: "iptables" or "ebpf"
     redirectMode: "iptables"
+    # Set ambient config dir path: defaults to /etc/ambient-config
+    cniAmbientConfigDir: ""
```
Comment (howardjohn) [inline manifests/charts/istio-cni/values.yaml:58]:
```suggestion
    configDir: ""
```

**Review by hanxiaop — APPROVED** [review]

File: releasenotes/notes/45404.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: feature
+area: cni
```
Comment (hanxiaop) [inline releasenotes/notes/45404.yaml:3]:
```suggestion
area: installation
```

Comment (rokkiter) [reply releasenotes/notes/45404.yaml]: thanks

**Review by howardjohn — APPROVED** [review]

File: istioctl/cmd/testdata/vmconfig/simple/hosts
Comment (hanxiaop) [inline istioctl/cmd/testdata/vmconfig/simple/hosts:1]: This should be extra change which result in the test failure. Please delete this file.

Comment (rokkiter) [reply istioctl/cmd/testdata/vmconfig/simple/hosts:1]: yes， I am try to fix it

**Pull request merged**